visualdl/server/lib.py

```diff
@@ -393,3 +392,23 @@ def _handler(key, func, *args, **kwargs):
         return data
 
     return _handler
+
+
+# A simple PCA implementaiton to do the dimension reduction.
+def simple_pca(x, dimension):
+    # Center the data.
+    x -= np.mean(x, axis=0)
+
+    # Computing the Covariance Matrix
+    cov = np.cov(x, rowvar=False)
+
+    # Get eigenvectors and eigenvalues from the covariance matrix
+    eigvals, eigvecs = np.linalg.eig(cov)
```

Comment:
What a math guy, Jeff! But do we need to do it ourselves? Moreover, SVD is more stable than eigenvector in terms of computing principle components. 

Check [here](https://stats.stackexchange.com/questions/314046/why-does-andrew-ng-prefer-to-use-svd-and-not-eig-of-covariance-matrix-to-do-pca)

Reply:
The reason behind it so we don't have to import another pip package to do the calculation. 
And actually the TSNE file already has a clean PCA implementation. and I could just use that.